core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1209,38 +1227,57 @@ private void updateMediaSession(final PlayerStatus playerStatus) {
 
         sessionState.setActions(capabilities);
 
-        // On Android Auto, custom actions are added in the following order around the play button, if no default
-        // actions are present: Near left, near right, far left, far right, additional actions panel
-        PlaybackStateCompat.CustomAction.Builder rewindBuilder = new PlaybackStateCompat.CustomAction.Builder(
+        if (UserPreferences.showRewindOnFullNotification()) {
+            // On Android Auto, custom actions are added in the following order around the play button, if no default
+            // actions are present: Near left, near right, far left, far right, additional actions panel
+            PlaybackStateCompat.CustomAction.Builder rewindBuilder = new PlaybackStateCompat.CustomAction.Builder(
```

Comment:
Making this conditional feels like it could mess up the order in which the buttons are shown on Android Auto (eg. fast forward on the left of the play button). I'm not sure how to best keep them in the expected place. Maybe by always showing rewind and fast forward, and making only the other ones configurable?

Reply:
Hm, do we have any idea how is order of the buttons determined? I'm not too familiar with the Android Auto.

I could remove the extra buttons, but I would prefer to keep them 100% configurable.

Reply:
I was the one who originally wrote that comment in PlaybackService and it pretty much describes how Android Auto (and the Android 13 notification) works. I have some screenshots from when I worked on the custom actions in PR #6050. ByteHamster is correct here - this system can result in some odd button placements on Android 13. For example, selecting "Fast forward", "Skip episode", and "Playback speed" leads to Fast Forward on the left, which means it would also be on the left in Android Auto as well.

![Odd Button Layout](https://github.com/AntennaPod/AntennaPod/assets/42682253/2bf23a3f-be67-4ddf-91da-e095529cf503)

I do think making sure these can be 100% configurable would be a good thing. I'm wondering if we could represent the setting to the user as a collection of five buttons in a defined order and not just a binary choice for each item? So the user could choose to (for example) swap "Playback speed" for "Next chapter" as the second-to-last item and the app would have to make sure it adds the correct actions to the PlaybackStateCompat in the right order.

Reply:
Thanks for joining in, @awbooze!

> I'm wondering if we could represent the setting to the user as a collection of four buttons in a defined order

You mean having 4 settings where one can assign every slot individually? Or one setting with some choices for different selections of buttons?

Reply:
I would think that a single setting where you could assign each slot would be best, but if that's too complex, I would go with something more like your first idea. I wouldn't want to have a couple predefined selections for the button layout and no way to create your own. I don't think most people will want to shift the "Fast forward" or "Skip backward" items around, but I do think a lot of people would want to replace or move around the other two if we add more options like this.

Reply:
> I would think that a single setting where you could assign each slot would be best, but if that's too complex, I would go with something more like your first idea.

Aren't these the same? The user has 4 things to change then. (how these are presented in the UI is a different thing)

> I don't think most people will want to shift the "Fast forward" or "Skip backward" items around

Then I would say that we do the following, as it is a simpler setting:

- Always show fast forward / fast rewind buttons
- Have a multi-select checkbox setting (similar to the one we have for "compact notification buttons) where users can select 2 additional settings. Default: Just "skip forward".
- If "skip forward" is selected, place it on the right. Then place the other items in arbitrary order.

Reply:
> Aren't these the same? The user has 4 things to change then. (how these are presented in the UI is a different thing)

Yeah, they're basically the same idea.

> Then I would say that we do the following, as it is a simpler setting:

That would be simpler, but I wouldn't want to start with just "Skip Forward". That would work well on the Android 13 notification but not on Android Auto, since that would put the "Skip Forward" button on the left. If we'd want to put it on the right we'd need a "Spacer" item, and then that would cause the Android 13 notification to have an extra space between actions. If we're going for easy, it would be easier to require people to select exactly two additional options.

Reply:
> If we're going for easy, it would be easier to require people to select exactly two additional options.

Sorry for the late reply. Okay, let's do that for now. Then the code setting up the buttons should make sure that if "skip" is selected, it should be the second action, so it is displayed on the right. Default would then be the current behavior, eg. skip and speed.

Reply:
I think I implemented those changes now, please review